src/main.cpp

```diff
@@ -6062,23 +6068,41 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
     }
 
 
-    else if (strCommand == NetMsgType::XPEDITEDREQUEST)
+    else if (strCommand == NetMsgType::XPEDITEDREQUEST && IsThinBlocksEnabled())
     {
+        if (!pfrom->ThinBlockCapable())
+        {
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
+            return error("XPEDITEDREQUEST message received from a non thinblock node, peer=%d", pfrom->GetId());
+        }
```

Comment:
I thought about the idea of such a policy, but didn't come up with any good solution except to move this policy into the hands of the node operator in the form of tweakable config:

https://bitco.in/forum/threads/buip053-tweakable-client-dos-responses.2074/